packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import {
+  css,
+  autobind,
+  BaseComponent
+} from '../../Utilities';
+import { IButtonProps } from '../../Button';
+import { IOverflowSetProps } from './OverflowSet.Props';
+import { FocusZone, FocusZoneDirection } from '../../FocusZone';
+
+import * as stylesImport from './OverflowSet.scss';
+const styles: any = stylesImport;
+
+export class OverflowSet extends BaseComponent<IOverflowSetProps, null> {
+
+  public render() {
+    let {
+      items,
+      overflowItems,
+      onRenderOverflowButton
+    } = this.props;
+
+    const overflowButtonProps: IButtonProps = {
+      menuProps: { items: overflowItems }
+    }
+
+    return (
+      <FocusZone className={ css('ms-OverflowSet', styles.root) } direction={ FocusZoneDirection.horizontal } role='menubar' >
+        { items && this._onRenderItems(items) }
+        { overflowItems.length && onRenderOverflowButton(overflowButtonProps) }
+      </FocusZone>
+    );
+  }
+
+  @autobind
+  private _onRenderItems(items: any[]): JSX.Element[] {
+    return items.map((item, i) => {
+      return (
+        <div key={ i } className={ css('ms-OverflowSet-item', styles.item) }>
```

Comment:
This isn't a great way to assign keys since it wouldn't let React's reconciler detect reorderings and insertions/deletions. More details can be found here https://facebook.github.io/react/docs/lists-and-keys.html

Can we provide a way for the consumer to pass in a unique identifier for each item?